.github/workflows/pre-commit.yml

```
@@ -74,12 +81,9 @@ jobs:
             golangci-lint.cache-${{ env.INTERVAL_KEY }}-
             golangci-lint.cache-
 
-      - name: Set up pre-commit
-        run: pipx install "pre-commit${PRE_COMMIT_VERSION}"
-
       - name: Set pre-commit cache key parts
         run: |
-          echo "GO_VERSION=$(<.go-version)" >> $GITHUB_ENV
+          echo "GO_VERSION=$(go version)" >> $GITHUB_ENV
```

Let's do that in 2 steps to avoid masking the exit code of `go version` if it fails

```suggestion
          GO_VERSION="$(go version)"
          echo "GO_VERSION=${GO_VERSION}" >> $GITHUB_ENV
```

https://www.shellcheck.net/wiki/SC2312